packages/api/src/__tests__/useSubscription.spec.tsx

```diff
@@ -1,86 +1,55 @@
+import { renderHook, act } from '@testing-library/react-hooks';
 import { useWS as useWSShared } from '@deriv/shared';
 import useSubscription from '../useSubscription';
-import { TSocketSubscribableEndpointNames, TSocketAcceptableProps } from '../../types';
 
 jest.mock('@deriv/shared');
 
 const mockUseWSShared = useWSShared as jest.MockedFunction<typeof useWSShared>;
 
-const UseSubscriptionExample = <T extends TSocketSubscribableEndpointNames>({
-    name,
-    request,
-}: {
-    name: T;
-    request?: TSocketAcceptableProps<T>;
-}) => {
-    const WS = useSubscription(name);
-
-    return (
-        <React.Fragment>
-            <p data-testid={'dt_is_loading'}>{WS.is_loading ? 'true' : 'false'}</p>
-            <p data-testid={'dt_error'}>{WS.error ? JSON.stringify(WS.error) : 'undefined'}</p>
-            <p data-testid={'dt_data'}>{WS.data ? JSON.stringify(WS.data) : 'undefined'}</p>
-            <button data-testid={'dt_subscribe'} onClick={() => WS.subscribe(request)}>
-                subscribe
-            </button>
-        </React.Fragment>
-    );
-};
-
 describe('useSubscription', () => {
     test('should subscribe to p2p_order_info and get the order updates', async () => {
         mockUseWSShared.mockReturnValue({
             subscribe: jest.fn(() => {
                 return {
-                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                    subscribe: async (onData: (response: any) => void, onError: (response: any) => void) => {
+                    subscribe: async (onData: (response: unknown) => void, onError: (response: unknown) => void) => {
                         const delay = (ms: number) => new Promise<never>(resolve => setTimeout(resolve, ms));
-
                         await delay(500);
-
                         onData({ p2p_order_info: { status: 'pending' } });
-
                         await delay(500);
-
                         onData({ p2p_order_info: { status: 'buyer-confirmed' } });
-
                         await delay(500);
-
                         onData({ p2p_order_info: { status: 'disputed' } });
-
                         await delay(500);
-
                         onError({ error: { code: 'Foo', message: 'Error message' } });
-
                         await delay(500);
-
                         onData({ p2p_order_info: { status: 'completed' } });
-
                         return { unsubscribe: () => Promise.resolve() };
                     },
                 };
             }),
         });
 
-        render(<UseSubscriptionExample name={'p2p_order_info'} request={{ id: '2' }} />);
+        const { result, waitForNextUpdate } = renderHook(() => useSubscription('p2p_order_info'));
+
+        expect(result.current.is_loading).toBe(false);
+        expect(result.current.error).toBe(undefined);
+        expect(result.current.error).toBe(undefined);
+        expect(result.current.error).toBe(undefined);
```

Comment:
```suggestion
        expect(result.current.error).toBe(undefined);
```
@hamid-deriv One should be enough, Right? 🤔

Reply:
Of course!!!

Reply:
Of course!!!